Models/ProjectModels/Validation/Data/User/Validator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+declare(strict_types=1);
+
+namespace Models\ProjectModels\Validation\Data\User;
+
+use Models\AbstractProjectModels\Validation\Data\User\Validator as BaseValidator;
+
+class Validator extends BaseValidator
+{
+    /**
+     * @param array $data
+     * @return array
+     */
+    public function correctCheck(array $data): array
+    {
+        $resultData = [];
+        foreach ($data as $key => $value) {
+            switch ($key) {
+                case 'login' :
+                case 'pass' :
+                    $resultData[$key] = $this->pregMatchStrLen('/[A-Za-z0-9]{4,16}/u', $value);
+                    break;
+                case 'pass_confirm' :
+                    if ($value !== $data['pass']) {
+                        $resultData[$key] = '';
```

Comment:
You can throw an exception here with a following text like "Password doesn't match" and catch it wherever you want.

Reply:
This is an example how it could be written:
``` 
public function correctCheck(array $data): array
    {
        $resultData = [];
        foreach ($data as $key => $value) {
            switch ($key) {
                case 'login' :
                case 'pass' :
                    $resultData[$key] = $this->pregMatchStrLen('/[A-Za-z0-9]{4,16}/u', $value);
                    break;
                case 'pass_confirm' :
                    if ($value !== $data['pass']) {
                        throw new \Exception('Passwords don't match!');
                      // or throw your exception class instance,
                    }
............

And now let's look at 
\Controllers\AbstractControllers\AbstractUserController::newAction
public function newAction(array $params = null): void
    {
        if ($this->userModel->isSigned()) {
            $this->homeLocationByCustomerType();
            return;
        }

        if (!$this->postInfo->isPost()) {
            $this->redirectHomeLocation();
            return;
        }
     
        try {
                     $emptyResult = $this->dataValidator->emptyCheck($this->postInfo->getPostData(), 'registration');
        if (in_array(false, $emptyResult)) {
            $this->checkResult($emptyResult, 'empty_data', 'registration');
        } else {
            $correctResult = $this->dataValidator->correctCheck($this->postInfo->getPostData());
            if (in_array('', $correctResult)) {
                $this->checkResult($correctResult, 'wrong_data', 'registration');
            } else {
                if ($this->fileInfo->isImageSent() && !$this->imageValidator->validate($params[0])) {
                    $this->checkResult($this->imageValidator->getErrors(), 'wrong_data', 'registration');
                } else {
                    if (!$this->userModel->newUser($params[0], $correctResult)) {
                        $this->redirectLocation('user/registration');
                    } else {
                        $this->redirectHomeLocation();
                    }
                }
            }
        }
     } catch (\Exception (or your exception class) $e) {
            // log error, or simply return, or do anything else.
     } 
    }
```